frontend_app_file/src/component/Kanban.jsx

```diff
@@ -0,0 +1,386 @@
+            onClick={() => this.removeCard(card)}
+          />
+        </div>
+        <div
+          className='file__contentpage__statewrapper__kanban__card__description'
+          onClick={() => this.handleEditCardContent(card)}
+        >
+          {card.description}
+        </div>
+      </div>
+    )
+  }
+
+  handleAddCardClicked (column) {
+    const title = prompt('Please enter the title of the card')
+    if (title) {
+      const description = prompt('Please enter the content of the card')
+
+      this.addCardToColumn(column, {
+        id: uuidv4(),
+        title,
+        description
+      })
+    }
+  }
+
+  addCardToColumn (column, card) {
+    this.updateColumns(
+      this.state.board.columns.map(
+        col => (
+          col === column
+            ? { ...col, cards: [...col.cards, card] }
+            : col
+        )
+      )
+    )
+  }
+
+  handleSaveClicked = async () => {
+    const { props } = this
+    const fetchResultSaveKanban = await handleFetchResult(
+      await putFileContent(
+        props.config.apiUrl,
+        props.content.workspace_id,
+        props.content.content_id,
+        props.content.label,
+        JSON.stringify(this.state.board)
+      )
+    )
+
+    // TODO factorize with the code doing the same thing in html doc
+    switch (fetchResultSaveKanban.apiResponse.status) {
+      case 200: {
+        removeLocalStorageItem(
+          'rawContent',
+          props.content,
+          'kanban'
+        )
+
+        this.setState({ mustSave: false, isDrafAvailable: false })
+        // TODO handle VIEW / edit mode
+        // TODO handle notify all
+        break
+      }
+      case 400:
+        switch (fetchResultSaveKanban.body.code) {
+          case 2044:
+            alert(props.t('You must change the status or restore this kanban board before any change'))
+            break
+          default:
+            alert(props.t('Error while saving the new version'))
+            break
+        }
+        break
+      default:
+        alert(props.t('Error while saving the new version'))
+        break
+    }
+  }
+
+  handleAddColumnClicked = (board, column) => {
+    // console.log("handleAddColumnClicked", board, column)
+    const title = prompt('Please enter the name of the new column')
+    if (!title) {
+      return
+    }
+
+    this.handleNewColumnConfirm({
+      title: title,
+      cards: []
+    })
+  }
+
+  handleCardRemove = (card) => {
+    // console.log("handleCardRemove", card)
+    this.removeCard(card)
+  }
+
+  handleColumnRemove = (column) => {
+    // console.log("handleColumnRemove", column)
+    this.updateColumns(this.state.board.columns.filter(col => col !== column))
+  }
+
+  handleCardDragEnd = (card, { fromColumnId, fromPosition }, { toColumnId, toPosition }) => {
+    // console.log("handleCardDragEnd", card, { fromColumnId, fromPosition }, {toColumnId, toPosition })
+    this.updateColumns(
+      this.state.board.columns.map(
```

Comment:
to reduce this complexity, this data structure might be interesting: https://medium.com/javascript-in-plain-english/https-medium-com-javascript-in-plain-english-why-you-should-use-an-object-not-an-array-for-lists-bee4a1fbc8bd

Reply:
or maybe decomposing the function

My point is, I wouldn't like to debug this function :sweat_smile:

Reply:
I can't use another data structure here, this is the one of the kanban component. But sure I can split this function :-)

Reply:
I can't use another data structure here, this is the one of the kanban component. But sure I can split this function :-)